sdk/core/azure-core/inc/azure/core/exception.hpp

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Define RequestFailedException. It is used by HTTP exceptions.
```

Comment:
Do we plan to include all custom-defined exception types in this file? If so, update this comment to be more general and maybe rename the file to `exceptions.hpp`.

Will the user ever include this header directly? If not, it might be better to have this exception types defined where the `Request` object is, rather than a separate header file.

Will the user ever not want to include it?

Reply:
> Will the user ever not want to include it?

If never, and if we expect to only have one/few of these, it would be better to move them into the existing headers.

Reply:
We can refactor where do we want this in the future after the first API review maybe.
For now, I don't see a better header where to smuggle this exception

Reply:
Again, a suggestion, you can do in the future, but I agree with Ahson, I would rather start from plural here - after all, the way we organize to have multiple related things in one header, it is more likely to start adding more exception-related stuff here, than keeping a single item.